azurerm/internal/services/storagecache/registration.go

```diff
@@ -0,0 +1,24 @@
+package storagecache
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+type Registration struct{}
+
+// Name is the name of this Service
+func (r Registration) Name() string {
+	return "Storage Cache"
+}
+
+// SupportedDataSources returns the supported Data Sources supported by this Service
+func (r Registration) SupportedDataSources() map[string]*schema.Resource {
+	return map[string]*schema.Resource{}
+}
+
+// SupportedResources returns the supported Resources supported by this Service
+func (r Registration) SupportedResources() map[string]*schema.Resource {
+	return map[string]*schema.Resource{
+		"azurerm_hpc_cache": resourceArmHPCCache(),
```

Comment:
rather than making this it's own service could we move this into the Storage service, since it forms part of that?